tools/buildgen/plugins/check_attrs.py

```diff
@@ -0,0 +1,124 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Buildgen attribute validation plugin."""
+
+
+def anything():
+  return lambda v: True
+
+
+def one_of(values):
+  return lambda v: v in values
+
+
+def subset_of(values):
+  return lambda v: all(e in values for e in v)
+
+
+VALID_ATTRIBUTE_KEYS_MAP = {
+    'filegroup': {
+        'deps': anything(),
+        'headers': anything(),
+        'plugin': anything(),
+        'public_headers': anything(),
+        'src': anything(),
+        'uses': anything(),
+    },
+    'lib': {
+        'baselib': anything(),
+        'boringssl': one_of((True, )),
+        'build_system': anything(),
+        'build': anything(),
+        'defaults': anything(),
+        'deps_linkage': one_of(('static',)),
+        'deps': anything(),
+        'dll': one_of((True, 'only')),
+        'filegroups': anything(),
+        'generate_plugin_registry': anything(),
+        'headers': anything(),
+        'language': one_of(('c', 'c++', 'csharp')),
+        'LDFLAGS': anything(),
+        'platforms': subset_of(('linux', 'mac', 'posix', 'windows')),
+        'public_headers': anything(),
+        'secure': anything(),
+        'src': anything(),
+        'vs_proj_dir': anything(),
+        'zlib': one_of((True, )),
+    },
+    'target': {
+        'args': anything(),
+        'benchmark': anything(),
+        'boringssl': one_of((True, )),
+        'build': anything(),
+        'ci_platforms': anything(),
+        'corpus_dirs': anything(),
+        'cpu_cost': anything(),
+        'defaults': anything(),
+        'deps': anything(),
+        'dict': anything(),
+        'exclude_configs': anything(),
+        'exclude_iomgrs': anything(),
+        'excluded_poll_engines': anything(),
+        'filegroups': anything(),
+        'flaky': one_of((True, False)),
+        'gtest': one_of((True, False)),
+        'headers': anything(),
+        'language': one_of(('c', 'c89', 'c++', 'csharp')),
+        'maxlen': anything(),
+        'platforms': subset_of(('linux', 'mac', 'posix', 'windows')),
+        'run': one_of((True, False)),
+        'secure': anything(),
+        'src': anything(),
+        'timeout_seconds:': anything(),
+        'timeout_seconds': anything(),
+        'uses_polling': anything(),
+        'vs_proj_dir': anything(),
+        'zlib': one_of((True, )),
+    },
+}
+
+
+def check_attributes(entity, type, errors):
+  attributes = VALID_ATTRIBUTE_KEYS_MAP[type]
+  name = entity.get('name', anything())
+  for key, value in entity.items():
+    if key == 'name':
+      continue
+    validator = attributes.get(key, None)
```

Comment:
Nit: `dict.get` returns `None` by default, so the second argument is superfluous.

Reply:
That's right!